src/app/api/user/email/route.ts

```diff
@@ -0,0 +1,49 @@
+import bcrypt from "bcrypt";
+import { response200, response400, response401 } from "@/app/utils";
+import { selectUserById, updateUserEmail } from "@/model/UserModel";
+import { z } from "zod";
+import getUserServerSide from "@/services/server/UserService";
+import { parseJsonFromBody } from "@/services/utils";
+
+const requestType = z.object({
+  email: z.string().email(),
+  password: z.string(),
+});
+
+export async function POST(request: Request) {
+  const user = await getUserServerSide();
+  if (user === undefined) {
+    return response401("User not logged in");
+  }
+
+  let body = await parseJsonFromBody(request);
```

Comment:
I would rename this. `body` is misleading here, because it's not actually the body.

Reply:
Also, why `let`?